starport/pkg/chaincmd/chaincmd.go

```diff
@@ -0,0 +1,252 @@
+	optionValidatorCommissionMaxRate       = "--commission-max-rate"
+	optionValidatorCommissionMaxChangeRate = "--commission-max-change-rate"
+	optionValidatorMinSelfDelegation       = "--min-self-delegation"
+	optionValidatorGasPrices               = "--gas-prices"
+
+	constTendermint = "tendermint"
+	constJSON       = "json"
+)
+
+type KeyringBackend string
+
+const (
+	KeyringBackendOS      KeyringBackend = "os"
+	KeyringBackendFile    KeyringBackend = "file"
+	KeyringBackendPass    KeyringBackend = "pass"
+	KeyringBackendTest    KeyringBackend = "test"
+	KeyringBackendKwallet KeyringBackend = "kwallet"
+)
+
+type ChainCmd struct {
+	appCmd         string
+	chainID        string
+	homeDir        string
+	keyringBackend KeyringBackend
+	cliCmd         string
+}
+
+// New creates a new ChainCmd to launch command with the chain app
+func New(appCmd string, options ...Option) ChainCmd {
+	chainCmd := ChainCmd{
+		appCmd: appCmd,
+	}
+
+	// Apply the options provided by the user
+	for _, applyOption := range options {
+		applyOption(&chainCmd)
+	}
+
+	return chainCmd
+}
+
+type Option func(*ChainCmd)
+
+// WithHome replaces the default home used by the chain
+func WithHome(home string) Option {
+	return func(c *ChainCmd) {
+		c.homeDir = home
+	}
+}
+
+// WithChainID provides a specific chain ID for the commands that accept this option
+func WithChainID(chainID string) Option {
+	return func(c *ChainCmd) {
+		c.chainID = chainID
+	}
+}
+
+// WithKeyrinBackend provides a specific keyring backend for the commands that accept this option
+func WithKeyrinBackend(keyringBackend KeyringBackend) Option {
+	return func(c *ChainCmd) {
+		c.keyringBackend = keyringBackend
+	}
+}
+
+// WithLaunchpadCLI provides the name of the CLI application to call Launchpad CLI commands
+func WithLaunchpadCLI(cliCmd string) Option {
+	return func(c *ChainCmd) {
+		c.cliCmd = cliCmd
+	}
+}
+
+// StartCommand returns the command to start the daemon of the chain
+func (c ChainCmd) StartCommand(options ...string) step.Option {
+	command := append([]string{
+		commandStart,
+	}, options...)
+	return step.Exec(c.appCmd, c.attachHome(command)...)
+}
+
+// InitCommand returns the command to initialize the chain
+func (c ChainCmd) InitCommand(moniker string) step.Option {
+	command := []string{
+		commandInit,
+		moniker,
+	}
+	command = c.attachChainID(command)
+	return step.Exec(c.appCmd, c.attachHome(command)...)
+}
+
+// AddKeyCommand returns the command to add a new key in the chain keyring
+func (c ChainCmd) AddKeyCommand(accountName string) step.Option {
+	command := []string{
+		commandKeys,
+		"add",
+		accountName,
+		optionOutput,
+		constJSON,
+	}
+	command = c.attachKeyringBackend(command)
+	return step.Exec(c.appCmd, c.attachHome(command)...)
+}
+
+// ImportKeyCommand returns the command to import a key into the chain keyring from a mnemonic
+func (c ChainCmd) ImportKeyCommand(accountName string) step.Option {
```

Comment:
Since `--recover` option is interactive, what do you think about implementing `ImportKeyCommand` like [this](https://github.com/tendermint/starport/blob/bab1919/starport/services/chain/plugin-stargate.go#L56-L69) by using `step.Write` to write interactive input?

But this will require `ImportKeyCommand` to return a `[]step.Option`.

Reply:
I prefer to keep minimal usage for this command and only returning `step.Option` of the command